rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -145,5 +236,120 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
+  } else {
+    gzerr << "Could not open custom wind field text file.\n";
+  }
+
+}
+
+/*
+  Linear interpolation
+    position: y-coordinate of the target point
+    values:   a pointer to an array of size 2 containing the values of the two points to interpolate from (12 and 13)
+    points:   a pointer to an array of size 2 containing the y-coordinate of the two points to interpolate from
+*/
+math::Vector3 GazeboWindPlugin::LinearInterpolation(double position, math::Vector3 *values, float *points) {
+  math::Vector3 value = values[0] + (values[1] - values[0]) / (points[1] - points[0]) * (position - points[0]);
+  return value;
+}
+
+/*
+  Bilinear interpolation
+    position: a pointer to an aray of size 2 containing the x and y-coordinates of the target point
+    values:   a pointer to an array of size 4 containing the values of the four points to interpolate from (8, 9, 10 and 11)
+    points:   a pointer to an array of size 6 containing the x-coordinate of the four points to interpolate from,
+              and the y-coordinate of the two intermediate points (12 and 13)
+*/
+math::Vector3 GazeboWindPlugin::BilinearInterpolation(double *position, math::Vector3 *values, float *points) {
+  math::Vector3 intermediate_values[2] = {
+                                            LinearInterpolation(position[0],&(values[0]),&(points[0])),
+                                            LinearInterpolation(position[0],&(values[2]),&(points[2]))
+                                         };
+  math::Vector3 value = LinearInterpolation(position[1], intermediate_values, &(points[4]));
+  return value;
+}
+
+/*
+  Trilinear interpolation
+    link_position:  a Vector3 containing the x, y and z-coordinates of the target point
+    values:         a pointer to an array of size 8 containing the values of the eight points to interpolate from (0, 1, 2, 3, 4, 5, 6 and 7)
+    points:         a pointer to an array of size 14 containing the z-coordinate of the eight points to interpolate from,
+                    the x-coordinate of the four intermediate points (8, 9, 10 and 11), and the y-coordinate of the last two intermediate points (12 and 13)
+*/
+math::Vector3 GazeboWindPlugin::TrilinearInterpolation(math::Vector3 link_position, math::Vector3 *values, float *points) {
+  double position[3] = {link_position.x,link_position.y,link_position.z};
+  math::Vector3 intermediate_values[4] = {
+                                            LinearInterpolation(position[2],&(values[0]),&(points[0])),
+                                            LinearInterpolation(position[2],&(values[2]),&(points[2])),
+                                            LinearInterpolation(position[2],&(values[4]),&(points[4])),
+                                            LinearInterpolation(position[2],&(values[6]),&(points[6]))
+                                         };
+  math::Vector3 value = BilinearInterpolation(&(position[0]),intermediate_values,&(points[8]));
```

Comment:
About passing the points[]- values here: I do agree that you need to pass the first 8 elements of this array, i.e. the z-coordinates of the 8 corner points, because in z-direction our grid is not equidistant and can also be sheared (as it follows the terrain). However, I think you don't need elements 8-14 _if_ you quickly recalculate the link_position.x and .y into local coordinates (0...1) inside that equidistant and rectangular grid in x,y-direction (thing is: You know that all points there can only have the value 0 or 1 and your order of points pre-defines what values they have). I hope you know what I mean.

This might help you save a bit more code size. But you would need to look at whether it's worth changing this. It might be more complicated/confusing to do it in this other way, so you decide.